app/services/search/location_builder.rb

```diff
@@ -44,7 +44,7 @@ def initialize_polygon_boundaries
 
     @polygon_boundaries = []
     locations.compact.each do |location|
-      polygons = buffer_radius.present? ? location.buffers[buffer_radius] : location.polygons["polygons"]
+      polygons = buffer_radius.present? ? location.buffers[buffer_radius] : location.buffers[radius.to_s]
```

Comment:
We might want to rethink our separation of 'buffer_radius' and 'radius'. It looks like `buffer_radius` is what we call the radius when it comes from wider search suggestions after a no-result search; and I don't think we want to do anything differently in the search using the `buffer_radius` parameter compared to using the `radius` parameter (so, they 'mean' the same thing).

If that's correct, I'm happy for that refactor to be a later PR.

Reply:
I've actually removed buffer_radius in another branch!

Reply:
In that case don't worry about my comments about the tests needing to test the cases when only one of radius and buffer_radius is present, as those tests will be short-lived.